checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleExecutor.java

```diff
@@ -61,14 +61,24 @@ public CheckstyleExecutor(CheckstyleConfiguration configuration, CheckstyleAudit
    * Execute Checkstyle and return the generated XML report.
    */
   public void execute() {
-    TimeProfiler profiler =
-            new TimeProfiler().start("Execute Checkstyle " + CheckstyleVersion.getVersion());
-    ClassLoader initialClassLoader = Thread.currentThread().getContextClassLoader();
-    Thread.currentThread().setContextClassLoader(PackageNamesLoader.class.getClassLoader());
-    URLClassLoader projectClassloader = createClassloader();
 
     Locale initialLocale = Locale.getDefault();
     Locale.setDefault(Locale.ENGLISH);
+    ClassLoader initialClassLoader = Thread.currentThread().getContextClassLoader();
+    Thread.currentThread().setContextClassLoader(PackageNamesLoader.class.getClassLoader());
+    try (URLClassLoader projectClassloader = createClassloader()) {
+      executeWithClassLoader(projectClassloader);
+    } catch (IOException e) {
+      throw new CheckstyleSensorException("could not close classloader", e);
```

Comment:
`executeWithClassLoader` does not throw a checked exception and neither does `createClassloader`. So the `IOException` can only come from the `close` method of `URLClassLoader`. The method is ultimately called by `org.sonar.api.batch.Sensor.analyse(Project, SensorContext)` which does not declare a checked exception so we have to do something here. Other options include:

1. One option would be to simply log a warning. Since the analysis was successful and only the closing of the class loader failed it can be argued that the analysis should complete successfully.
1. An other option is to use an existing unchecked exception or even `RuntimeException` directly, `CheckstyleException` is a checked exception.

In have no problem with either of these.